src/app/shapes/ShapesLibraryPlugin.ts

```
@@ -0,0 +1,170 @@
+import * as Konva from "konva";
+import { Shape } from "./Shape";
+import { Rectangle } from "./Rectangle";
+import { Ellipse } from "./Ellipse";
+import { Triangle } from "./Triangle";
+import { Line } from "./Line";
+import { TextShape } from "./TextShape";
+
+export class ShapesLibraryPlugin implements IShapesLibraryPlugin {
+    private nameGenerationCounter: number;
+    private shapes: {[name: string] : Shape };
+
+    private layer: Konva.Layer;
+    private stage: Konva.Stage;
+
+    public constructor(layer: Konva.Layer, stage: Konva.Stage) {
+        this.nameGenerationCounter = 0;
+        this.shapes = {};
+        this.layer = layer;
+        this.stage = stage;
+    }
+
+    private generateName(shapeType: string): string {
+        this.nameGenerationCounter++;
+        return shapeType + this.nameGenerationCounter.toString();
+    }
+
+    private addShape(shapeType: string, shapeProvider: (name: string) => Shape): string {
+        const name = this.generateName(shapeType);
+        const shape = shapeProvider(name);
+        this.shapes[name] = shape;
+
+        this.layer.add(shape.instance);
+        this.stage.draw();
+
+        return name;
+    }
+
+    public addRectangle(width: number, height: number): string {
+        return this.addShape("Rectangle", (name) => new Rectangle(name, width, height));
+    }
+
+    public addEllipse(width: number, height: number): string {
+        return this.addShape("Ellipse", (name) => new Ellipse(name, width, height));
+    }
+
+    public addTriangle(x1: number, y1: number, x2: number, y2: number, x3: number, y3: number): string {
+        return this.addShape("Triangle", (name) => new Triangle(name, x1, y1, x2, y2, x3, y3));
+    }
+
+    public addLine(x1: number, y1: number, x2: number, y2: number): string {
+        return this.addShape("Line", (name) => new Line(name, x1, y1, x2, y2));
+    }
+    
+    public addImage(): string {
```

Actually, @MicahElizabeth should we have just renamed the variable from `imageName` to `_` to make the compiler ignore the error, and still completely implement the interface?

---

I didn't get an error about an incomplete interface, so we're probably okay. I can do that if y'all don't mind doing the approval cycle again.

---

You don't need to. This is still implementing the interface. Arguments will be ignored when this function is called.

---

Go ahead and do it, just so someone looking at in the future would know. Sorry I didn't think of it before!